### DISCUSSION TITLE
Users are displayed to be added to groups even if they are already in the group

### DISCUSSION THREAD

**Discussion by notfounnd** [outline/outline#7875] [Self Hosting] [answered]
I’m following the guidance of the documentation [[link](https://docs.getoutline.com/s/guide/doc/users-groups-cwCxXP8R3V#h-groups)] for maintenance of collections by groups and I noticed that even if users have already been added to groups they appear again to be added when I will include some new member.

For example:

- I currently have 48 users in my database.
![image](https://github.com/user-attachments/assets/557717b4-96ac-4ac6-9806-094589a548af)

- I have a group that has 45 members (general).
![image](https://github.com/user-attachments/assets/70272345-6ffd-4a73-8670-e58d7fe3e9ad)

- When access to the screen to add the 3 new members are displayed users who are already in the group.
![image](https://github.com/user-attachments/assets/0d3a014d-b3ca-4e96-94f4-b8da6c76c9f1)

Any guidance to solve this? I'm using v0.80.2.

**Comment by tommoor** [comment, marked as answer]
Looks like it's a bug when there are more than 25 people in the group, will address for next version

**Comment by notfounnd** [comment]
@tommoor tks! 🚀

In time, I noticed some records getting duplicated in the database while I was researching about the problem I reported.

I was using version 0.79.1 when I noticed this.

The detail is that I went through many hypotheses studying the tool until reporting the problem in this thread. I don’t remember which table I saw this happening.

Steps:
- Add user in a group (this created record in the table)
- Remove user from the same group (this does not remove the record in the table)
- Add the user again in the group (this creates a new record in the table)

After tuning the tool to the latest version I could not find this data and nor reproduce again the problem.

**Reply by tommoor** [reply]
Previously `GroupUser` was paranoid, so it would just write a `deletedAt` column instead of removing the row – this is no longer the case as of https://github.com/outline/outline/pull/7356, so this is likely the "duplicate" you were seeing
